docs/canary_releases.md

```diff
@@ -0,0 +1,70 @@
+# Canary releases
+
+There is a need to use canary releases in order to make sure newer versions
+of the gathering conditions don't break the Insights Operator gatherer.
+
+This is how a canary release would look like:
+
+```mermaid
+sequenceDiagram
+    actor RT as Rules Team
+    actor PT as Processing Team
+    participant GC as Gathering Conditions
+    participant AI as App Interface
+    participant GCS as Gathering Conditions Service
+    participant Unleash
+    participant Storage
+    actor C as Cluster
+    actor CC as Canary Cluster
+
+    Note left of GCS: Latest version is 1.0.1
+    GCS -->> C: Get version 1.0.1
+    GCS -->> CC: Get version 1.0.1
+
+    RT ->> GC: Push a new tag 1.0.2
+    RT ->> AI: Set the canary and stable versions<br/>CANARY_VERSION = 1.0.2<br/>STABLE_VERSION = 1.0.1
+    PT ->> AI: Review and merge the PR
+    AI ->> GCS: Update prod deployment
+    
+    rect rgb(250,128,114)
+        alt canary or stable version not found
+            GCS ->> GCS: CrashLoopBackOff
+            PT ->> AI: Revert the PR
+        end
+    end
+
+    GCS -->> C: Get version 1.0.1
+    GCS -->> CC: Still get version 1.0.1
+
+    RT ->> Unleash: Increase the canary population to X%
+    GCS -->> C: Get version 1.0.1
+    rect rgb(143,188,143)
+        GCS -->> CC: Get version 1.0.2
+    end
+    
+    RT ->> Storage: Analyze the canary archives and check they are generated well
+
+    alt canary archives are wrong
+        RT ->> AI: Revert the PR
```

Comment:
If you revert the PR, the canary version will be `none`? Then what happens with the population label as "canary"?

Reply:
The happy day scenario below also has a period of time where `CANARY_VERSION = none` while some clusters still get the canary label. It suggests that the service serves the `STABLE_VERSION`.

Reply:
I've already answered [here](https://github.com/RedHatInsights/insights-operator-gathering-conditions-service/pull/234#discussion_r1677435200) but I'll copy here too:
> you keep the canary population to X% but remove the canary version: all the clusters, be it from the canary or live population, will consume the stable version

>The happy day scenario below also has a period of time where CANARY_VERSION = none while some clusters still get the canary label. It suggests that the service serves the STABLE_VERSION.

yes, exactly :)

Reply:
I think Iker's point was that it should be obvious from the diagram what the behavior is. [These changes](https://github.com/RedHatInsights/insights-operator-gathering-conditions-service/pull/234#discussion_r1675783452) should help with that.

Reply:
Yes, thanks.